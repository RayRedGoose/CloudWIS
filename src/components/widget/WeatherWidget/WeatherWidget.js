import React from 'react'
import './WeatherWidget.scss'
import WeatherImage from '../WeatherImage/WeatherImage'
import WeatherDisplay from '../WeatherDisplay/WeatherDisplay'

const WeatherWidget = (props) => {
  return (
    <section className="weather-widget">
      <WeatherImage type={props.weatherCurrentData.desc}/>
      <WeatherDisplay {...props} />
    </section>
  )
}

export default WeatherWidget
