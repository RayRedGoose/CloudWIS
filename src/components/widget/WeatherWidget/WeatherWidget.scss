.weather-widget {
  display: flex;
  background-color: #020202;
  border-radius: 20px;
  width: 85vw; height: 90vh;
  overflow: hidden;
}
