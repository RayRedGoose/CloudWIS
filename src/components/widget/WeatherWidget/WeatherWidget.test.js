import React from 'react';
import { shallow } from 'enzyme';
import WeatherWidget from './WeatherWidget';

it('should match the snapshot with all data passed in correctly', () => {
  const mockProps = {
    weatherCurrentData: {desc: 'current weather info'},
    weatherDaysData: {desc: 'five days weather info'},
    searchCity: 'Denver'
  }

  const widget = shallow(
    <WeatherWidget {...mockProps} />
  );

  expect(widget).toMatchSnapshot();
});
