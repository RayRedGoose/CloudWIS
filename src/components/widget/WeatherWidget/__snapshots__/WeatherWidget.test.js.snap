// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot with all data passed in correctly 1`] = `
<section
  className="weather-widget"
>
  <WeatherImage
    type="current weather info"
  />
  <WeatherDisplay
    searchCity="Denver"
    weatherCurrentData={
      Object {
        "desc": "current weather info",
      }
    }
    weatherDaysData={
      Object {
        "desc": "five days weather info",
      }
    }
  />
</section>
`;
